lib/plugins/aws/invokeLocal/index.js

```diff
@@ -377,58 +364,50 @@ class AwsInvokeLocal {
     const dockerfilePath = path.join('.serverless', 'invokeLocal', runtime, 'Dockerfile');
     const dockerfileCachePath = path.join(cachePath, 'dockerfiles', runtime, 'Dockerfile');
 
-    return fileExists(dockerfileCachePath)
-      .then(exists => {
-        if (exists) {
-          return fse.readFile(dockerfileCachePath).then(contents => {
-            if (contents.toString() === dockerFileContent) {
-              return this.checkDockerImage(imageName);
-            }
-            return false;
-          });
-        }
-        return false;
-      })
-      .then(cacheEquals => {
-        if (cacheEquals) {
-          return imageName;
-        }
-        return BbPromise.all([
-          ensureDir(path.join('.serverless', 'invokeLocal', runtime)),
-          ensureDir(path.join(cachePath, 'dockerfiles', runtime)),
-        ])
-          .then(() => {
-            this.serverless.cli.log('Writing Dockerfile...');
-            return BbPromise.all([
-              fse.writeFile(dockerfilePath, dockerFileContent),
-              fse.writeFile(dockerfileCachePath, dockerFileContent),
-            ]);
-          })
-          .then(() => {
-            this.serverless.cli.log('Building Docker image...');
-            return spawnExt('docker', [
-              'build',
-              '-t',
-              imageName,
-              `${this.serverless.config.servicePath}`,
-              '-f',
-              dockerfilePath,
-            ]).then(
-              () => {
-                return imageName;
-              },
-              ({ code, stdBuffer }) => {
-                if (stdBuffer) {
-                  process.stdout.write(stdBuffer);
-                }
-                throw new Error(`Failed to build docker image (exit code ${code}})`);
-              }
-            );
-          });
-      });
+    let cacheEquals = false;
+    const exists = await fileExists(dockerfileCachePath);
+    if (exists) {
+      const contents = await fse.readFile(dockerfileCachePath);
+
+      if (contents.toString() === dockerFileContent) {
+        cacheEquals = this.checkDockerImage(imageName);
```

Comment:
It misses `await`, also instead of creating `cacheEquals` it'll be nice to simply do:

```javascript
if (await this.checkDockerImage(imageName)) return imageName;
```